tensorflow-frameworks/tensorflow-data/README.md

```diff
@@ -0,0 +1,220 @@
+
```

Comment:
@dhruvrajan , there is a few typos in that README but before going through them, is this PR ready to be merged or it is still work in progress?

Reply:
I think it is ready to be merged, since it provides most of the structural features we want.

Reply:
I scanned the README; I hope it's a little freer of typos :)